### PR TITLE
Remove excess strings in roe naming

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
None

#### Purpose of change
When I added roe I wasn't aware that name strings can be carried via ``copy-from``. Now that I am aware, it's time to clean this up to lift some strain from translators.

### CODE REVIEW

**Comment by Karol1223** [comment]
Or uh. I guess not. Apparently name inheritance works only for monsters because of reasons.

Hey @Procyonae I know you had fun with your C++ method to cut down on strings for variants, want to help me out with this one as well?

**Comment by Uwuewsky** [comment]
> Now that I am aware, it's time to clean this up to lift some strain from translators.

Thanks, but don't worry about it, gettext uses the same translation for duplicate names
изображение

**Comment by Karol1223** [comment]
> Don't worry about it, gettext uses the same translation for duplicate names 

Ah I wasn't aware. Still, it'd be nice to throw the excess strings out to avoid naming issues in the future like there were in the initial implementation of marine fish roe.

File: data/json/items/comestibles/egg.json [hunk 1]
```diff
@@ -807,35 +762,31 @@
...
     "id": "egg_fallfish",
-    "name": { "str_sp": "roe" },
     "copy-from": "egg_fish",
     "rot_spawn": "GROUP_EGG_FALLFISH"
   },
   {
     "type": "COMESTIBLE",
     "abstract": "egg_fish_mutant",
-    "name": { "str": "roe" },
+    "name": { "str_sp": "mutant fish roe" },
```
Comment (Maleclypse) [inline data/json/items/comestibles/egg.json:789]:
```suggestion
    "name": { "str_sp": "mutant fish roe" }
```

File: data/json/items/comestibles/egg.json [hunk 2]
```diff
@@ -807,35 +762,31 @@
...
     "id": "egg_fallfish",
-    "name": { "str_sp": "roe" },
     "copy-from": "egg_fish",
     "rot_spawn": "GROUP_EGG_FALLFISH"
   },
   {
     "type": "COMESTIBLE",
     "abstract": "egg_fish_mutant",
-    "name": { "str": "roe" },
+    "name": { "str_sp": "mutant fish roe" }
```
Comment (Maleclypse) [inline data/json/items/comestibles/egg.json:789]:
```suggestion
    "name": { "str_sp": "mutant fish roe" },
```

**Comment by Karol1223** [comment]
...don't tell me the comma was all that was blocking this...

File: data/json/items/comestibles/egg.json [hunk 3]
```diff
@@ -490,315 +490,270 @@
   {
     "type": "COMESTIBLE",
     "id": "egg_brown_trout",
-    "name": { "str_sp": "roe" },
```
Comment (Maleclypse) [inline data/json/items/comestibles/egg.json:493]: Are you sure name isn't a required field?  Cause that error report sure seems to be suggesting it is.

**Comment by Maleclypse** [comment]
> ...don't tell me the comma was all that was blocking this...

No but there was another error that was preventing it from reading when I looked at what was going on with the tests and so in case that build was just bad I reset and restarted the test suite by making an edit and then undoing it.

File: data/json/items/comestibles/egg.json (hunk 3 above)
Comment (Karol1223) [reply data/json/items/comestibles/egg.json:493]: It is. Waiting for #71199

**Pull request closed without merging**